### PR TITLE
PLT-554 Small refactor of websocket code on client and server

### PR DESCRIPTION
**Pull request by jwilander**

### CODE REVIEW

File: web/react/stores/socket_store.jsx [hunk 1]
```diff
@@ -104,6 +144,138 @@ class SocketStoreClass extends EventEmitter {
...
 
+function handleNewPostEvent(msg) {
+    // Store post
+    const post = JSON.parse(msg.props.post);
+    PostStore.storePost(post);
+
+    // Update channel state
+    if (ChannelStore.getCurrentId() === msg.channel_id) {
+        if (window.isActive) {
+            AsyncClient.updateLastViewedAt();
```
Comment (crspeller) [inline web/react/stores/socket_store.jsx]: Why is this necessary?

Comment (jwilander) [inline web/react/stores/socket_store.jsx]: When the user is actively clicked on the channel that receives the post we want to update the last time they viewed the channel to now. So if they switch away from the channel and back, the new message indicator is not present for a message they've already seen

**Comment by crspeller** [comment]
+1

**Comment by crspeller** [comment]
@jwilander Needs rebase
